src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -54,14 +52,15 @@ public class PSVersionInfo
         /// For each later release of PowerShell, this constant needs to
         /// be updated to reflect the right version.
         /// </remarks>
-        private static Version s_psV1Version = new Version(1, 0);
-        private static Version s_psV2Version = new Version(2, 0);
-        private static Version s_psV3Version = new Version(3, 0);
-        private static Version s_psV4Version = new Version(4, 0);
-        private static Version s_psV5Version = new Version(5, 0);
-        private static Version s_psV51Version = new Version(5, 1, NTVerpVars.PRODUCTBUILD, NTVerpVars.PRODUCTBUILD_QFE);
-        private static SemanticVersion s_psV6Version = new SemanticVersion(6, 0, 0, null, null);
-        private static SemanticVersion s_psV7Version;
+        private static readonly Version s_psV1Version = new Version(1, 0);
+        private static readonly Version s_psV2Version = new Version(2, 0);
+        private static readonly Version s_psV3Version = new Version(3, 0);
+        private static readonly Version s_psV4Version = new Version(4, 0);
+        private static readonly Version s_psV5Version = new Version(5, 0);
+        private static readonly Version s_psV51Version = new Version(5, 1, NTVerpVars.PRODUCTBUILD, NTVerpVars.PRODUCTBUILD_QFE);
+        private static readonly SemanticVersion s_psV6Version = new SemanticVersion(6, 0, 0, null, null);
+        private static readonly SemanticVersion s_psCurrentVersion;
+        private static readonly Version s_psVersionCache;
```

Comment:
If we use psVersionCache we could rename psCurrentVersion to psSemanticVersionCache.

Reply:
I will update the member names.

Reply:
Done.